contribs/gnodev/pkg/browser/model_view.go

```diff
@@ -0,0 +1,174 @@
+package browser
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/charmbracelet/lipgloss"
+)
+
+var (
+	boxRoundedStyle = func(r *lipgloss.Renderer) lipgloss.Style {
+		b := lipgloss.RoundedBorder()
+		return r.NewStyle().
+			BorderStyle(b).
+			Padding(0, 2)
+	}
+
+	inputStyleLeft = func(r *lipgloss.Renderer) lipgloss.Style {
+		b := lipgloss.RoundedBorder()
+		b.Right = "├"
+		return r.NewStyle().
+			BorderStyle(b).
+			Padding(0, 2)
+	}
+
+	infoStyle = func(r *lipgloss.Renderer) lipgloss.Style {
+		b := lipgloss.RoundedBorder()
+		b.Left = "┤"
+		return boxRoundedStyle(r).Copy().BorderStyle(b)
+	}
+)
+
+func (m model) View() string {
+	if !m.bannerDiscarded {
+		return m.bannerView()
+	}
+
+	if !m.ready {
+		return "+"
+	}
+
+	mainView := fmt.Sprintf("%s\n%s\n%s", m.headerView(), m.bodyView(), m.footerView())
+	return m.zone.Scan(mainView)
+}
+
+func (m model) bannerView() string {
+	banner := m.banner.View()
+	if banner == "" || m.width == 0 || m.height == 0 {
+		return ""
+	}
+
+	// XXX: Encapsulate banner to avoid banner glitches
```

Comment:
What banner glitches? 👀

Reply:
These glitches occur if you do not take care to encapsulate the line with an ANSI reset sequence, which is naturally provided by the line below. If you don't do this, some ANSI escape sequences from a previous line may carry over to the next line, resulting in unexpected style output.